File: frontend/src/types/document.ts
export interface Document {
  id: string;
  title: string;
  content: any[];
  createdAt: Date;
  updatedAt: Date;
  icon?: string | null;
  parentId?: string | null;
  trashed: boolean;
  trashedAt?: Date | null;
  userId: string;
  workspaceId?: string; // Add workspace support
}

export interface DocumentStore {
  documents: Document[];
  activeDocumentId: string | null;
  isLoading: boolean;
  error: string | null;
}

export interface CreateDocumentInput {
  title: string;
  content: any[];
  icon?: string | null;
  parentId?: string | null;
  userId: string;
  workspaceId?: string; // Add workspace support
}

export interface UpdateDocumentInput {
  title?: string;
  content?: any[];
  icon?: string | null;
  parentId?: string | null;
  userId?: string;
  workspaceId?: string; // Add workspace support
}

File: backend-core-service/be-core/src/main/java/com/naammm/becore/entity/Document.java
package com.naammm.becore.entity;

import java.time.LocalDateTime;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.PrePersist;
import jakarta.persistence.PreUpdate;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "documents")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Document {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;

    @Column(nullable = false)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String content; // Markdown/Plain text content (NOT JSON)

    @Column(name = "user_id", nullable = false, length = 160)
    private String userId;

    @Column(name = "workspace_id", length = 160)
    private String workspaceId;

    @Column(nullable = false)
    private LocalDateTime createdAt;

    @Column(nullable = false)
    private LocalDateTime updatedAt;

    private String icon;

    @Column(name = "parent_id")
    private String parentId;

    @Column(nullable = false)
    private Boolean trashed;

    private LocalDateTime trashedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
        if (trashed == null) {
            trashed = false;
        }
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}
